homeassistant/components/assist_pipeline/websocket_api.py

```diff
@@ -273,3 +277,63 @@ def websocket_get_run(
         msg["id"],
         {"events": pipeline_runs[pipeline_run_id].events},
     )
+
+
+@callback
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "assist_pipeline/language/list",
+    }
+)
+@websocket_api.async_response
+async def websocket_list_languages(
+    hass: HomeAssistant,
+    connection: websocket_api.connection.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """List languages which are supported by a complete pipeline.
+
+    This will return a list of languages which are supported by at least one stt, tts
+    and conversation engine respectively.
+    """
+    conversation_manager = conversation.get_agent_manager(hass)
+    conversation_languages = set()
+    stt_languages = set()
+    tts_languages = set()
+    conversation_match_all = False
+
+    for agent_info in conversation_manager.async_get_agent_info():
+        agent = await conversation_manager.async_get_agent(agent_info["id"])
+        for language_tag in agent.supported_languages:
+            if language_tag == MATCH_ALL:
+                conversation_match_all = True
+                continue
+            dialect = language_util.Dialect.parse(language_tag)
+            conversation_languages.add(dialect.language)
+
+    stt_component: EntityComponent[stt.SpeechToTextEntity] = hass.data[stt.DOMAIN]
+    legacy_stt_providers: dict[str, stt.Provider] = hass.data[stt.DATA_PROVIDERS]
```

Comment:
We can't hook into `hass.data` of other integrations. STT should expose a function to get supported languages.